### PR TITLE
fix: add error handling when use customized config 

### PR DESCRIPTION
**Pull request by Yiyiyimu**
Signed-off-by: yiyiyimu <wosoyoung@gmail.com>

### What this PR does / why we need it:
Ref: #4337 

When implementing with #2101, no error handling is implemented so with the invalid config file, apisix would still start without the correct config setup. Here to fix it.

One thing to mention, since I failed to figure out how to get stderr separately when running shell command in Lua, considering normally stdout of `ls` and `mv` would always be empty (on ubuntu), I forward stderr to stdout to get the error msg and fail fast. Not sure if it's working in all OS.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/cli/util.lua [hunk 1]
```diff
@@ -47,6 +47,14 @@ function _M.execute_cmd(cmd)
     return data
 end
 
+_M.execute_cmd = execute_cmd
+
+-- For ls and mv which stdout would be always be empty,
```
Comment (spacewander) [inline apisix/cli/util.lua:52]: For commands which stdout

File: apisix/cli/ops.lua [hunk 2]
```diff
@@ -657,8 +658,18 @@ local function start(env, ...)
     if customized_yaml then
         profile.apisix_home = env.apisix_home .. "/"
         local local_conf_path = profile:yaml_path("config")
-        util.execute_cmd("mv " .. local_conf_path .. " " .. local_conf_path .. ".bak")
-        util.execute_cmd("ln " .. customized_yaml .. " " .. local_conf_path)
+
+        local err = util.execute_cmd_with_error("mv " .. local_conf_path .. " "
+                                                .. local_conf_path .. ".bak")
+        if len(err) > 0 then
```
Comment (spacewander) [inline apisix/cli/ops.lua:664]: Use `#err` is enough

Comment (Yiyiyimu) [reply apisix/cli/ops.lua]: Thx! Fixed

**Review by spacewander — APPROVED** [review]

**Review by spacewander — CHANGES_REQUESTED** [review]
CI failed.

**Review by tokers — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Comment by Yiyiyimu** [comment]
@spacewander PTAL

**Review by spacewander — APPROVED** [review]

**Pull request merged**